### DISCUSSION TITLE
SpringBootApplication can't start up with otel-javaagent v1.15.0 with opentelemetry-samplers.jar

### DISCUSSION THREAD

**Discussion by bgong-mdsol** [open-telemetry/opentelemetry-java-instrumentation#6234] [Q&A] [answered]
Hi,
we build a custom-otel-javaagent based on otel-javaagent, SpringBootApplication failed to start up when upgrade to otel-javaagent `v1.15.0` (see the error below):

> Error starting Tomcat context. Exception: org.springframework.beans.factory.UnsatisfiedDependencyException. Message: Error creating bean with name 'webMvcMetricsFilter' defined in class path resource [org/springframework/boot/actuate/autoconfigure/metrics/web/servlet/WebMvcMetricsAutoConfiguration.class]: Unsatisfied dependency expressed through method 'webMvcMetricsFilter' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'compositeMeterRegistry' defined in class path resource [org/springframework/boot/actuate/autoconfigure/metrics/CompositeMeterRegistryConfiguration.class]: Initialization of bean failed; nested exception is java.lang.LinkageError: loader constraint violation: loader 'bootstrap' wants to load interface io.opentelemetry.javaagent.shaded.io.opentelemetry.api.common.Attributes. A different interface with the same name was previously loaded by io.opentelemetry.javaagent.bootstrap.AgentClassLoader @7aec35a. (io.opentelemetry.javaagent.shaded.io.opentelemetry.api.common.Attributes is in unnamed module of loader io.opentelemetry.javaagent.bootstrap.AgentClassLoader @7aec35a, parent loader io.opentelemetry.javaagent.bootstrap.AgentClassLoader$JdkHttpServerClassLoader @300ffa5d)

I just simply included `opentelemetry-samplers` in [examples/distro](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/examples/distro/custom/build.gradle)
```
implementation("io.opentelemetry.contrib:opentelemetry-samplers:1.15.0-alpha")
```
[SpringBootSmokeTest](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/examples/distro/smoke-tests/src/test/java/com/example/javaagent/smoketest/SpringBootSmokeTest.java) failed.

Any idea or suggestion? Please let me know if you need a issue for this and which repo `opentelemetry-java-instrumentation` or `opentelemetry-java-contrib` should be used to open issue.

**Comment by laurit** [comment, marked as answer]
Looks like you have shaded `opentelemetry-api` in both boot loader and under `inst/` (where it shouldn't be). Try adding `implementation("io.opentelemetry.contrib:opentelemetry-samplers:1.15.0-alpha")` to `custom/build.gradle` also in the same file add 
```
    dependencies {
      exclude(dependency("io.opentelemetry:opentelemetry-api"))
      exclude(dependency("io.opentelemetry:opentelemetry-context"))
      exclude(dependency("io.opentelemetry:opentelemetry-semconv"))
    }
```
inside `shadowJar`. Hopefully this way you'll get classes from `opentelemetry-samplers` in `inst/` and won't get the classes that are supposed to be in boot loader.
